File: src/main/scala/com/github/mauricio/postgresql/messages/frontend/CredentialMessage.scala
package com.github.mauricio.postgresql.messages.frontend

import com.github.mauricio.postgresql.messages.backend.{Message, AuthenticationResponseType}

/**
 * User: mauricio
 * Date: 3/31/13
 * Time: 1:43 AM
 */
class CredentialMessage(
                         val username : String,
                         val password : String,
                         val authenticationType : AuthenticationResponseType.AuthenticationResponseType,
                         val salt : Option[Array[Byte]]
                         )
  extends FrontendMessage( Message.PasswordMessage )

File: postgresql-async/src/main/java/com/github/mauricio/async/db/postgresql/messages/frontend/CredentialMessage.kt
package com.github.mauricio.async.db.postgresql.messages.frontend

import com.github.mauricio.async.db.postgresql.messages.backend.AuthenticationResponseType
import com.github.mauricio.async.db.postgresql.messages.backend.ServerMessage


class CredentialMessage(val username: String,
                        val password: String,
                        val authenticationType: AuthenticationResponseType,
                        val salt: Array<Byte>?) : ClientMessage(ServerMessage.PasswordMessage)